src/main/java/org/jitsi/videobridge/DtlsTransport.java

```diff
@@ -380,6 +381,11 @@ protected void onIceConnected()
         TaskPools.IO_POOL.submit(() -> {
             try
             {
+                if (dtlsStack.getRole() == null)
+                {
+                    logger.warn(logPrefix +
```

Comment:
do things not throw when this happens?  i think they should, as nothing will work

Reply:
There's explicit code to handle the case (which fails silently if the role is never set): https://github.com/jitsi/jitsi-media-transform/blob/master/src/main/kotlin/org/jitsi/nlj/dtls/DtlsStack.kt#L171

Reply:
ah right, i think that does happen at times.  so why the warning then?

maybe we just want to add a timeout on that future and then log an error if the role isn't set within the timeout (probably better than waiting indefinitely)

Reply:
I don't think it should normally happen: clients should provide the necessarty info together with the ICE ufrag/password, and this executes after ICE has completed hence we should know the state. I see it with jigasi, which doesn't work right now, so I prefer to see a warning if it happens.

Agreed about having a timeout, will add a card